src/librustc_borrowck/borrowck/gather_loans/mod.rs

```diff
@@ -201,6 +219,19 @@ fn check_aliasability<'a, 'tcx>(bccx: &BorrowckCtxt<'a, 'tcx>,
     }
 }
 
+fn check_yields<'a, 'tcx>(bccx: &BorrowckCtxt<'a, 'tcx>,
```

Comment:
I would expect this check to be done in the `check_loans` code -- basically, each time we reach a yield point, we would check that no loans are in scope.